aws/src/integration/java/org/apache/iceberg/aws/s3/TestS3FileIOIntegration.java

```diff
@@ -445,6 +448,41 @@ public void testPrefixDelete() {
             });
   }
 
+  @Test
+  public void testFileRecoveryHappyPath() throws Exception {
+    S3FileIO s3FileIO = new S3FileIO(clientFactory::s3, new S3FileIOProperties());
+    s3.putBucketVersioning(
+        PutBucketVersioningRequest.builder()
+            .bucket(bucketName)
+            .versioningConfiguration(
```

Comment:
Just wrote these tests quickly to validate the implementation but will need to be more thoughtful here for people running these integration tests since now we'd be enabling bucket versioning.